js/instant_page.min.js

```diff
@@ -1,18 +1,28 @@
+	  if (result.active == false) {
+		console.log("[faster pageload plugin] Inactive: Do not load instant.page script")
+		return
+	  }
+	  if (result.preloadViewport == true) {
+		document.body.dataset.instantIntensity = "viewport-all";
+	  }
+	  
+	  if (!(arrayContains(window.location.host, result.blacklistSites.split(", ")))) {
```

Comment:
Please use the [happy path scheme](https://en.wikipedia.org/wiki/Happy_path) as it gives less indentation and creates a better to read code.

```
if (arrayContains(window.location.host, result.blacklistSites.split(", "))) { 
  return
}
```

Reply:
I think `result.blacklistSites.split(", ")` explodes when `result.blacklistSites` is `undefined`. Did you check that?

Reply:
How can I test is something explodes?

Reply:
I used this to help with a blank:
  if (result.blacklistSites !== "" && result.blacklistSites.split(", ").includes(window.location.host)){
I think it should work because apparently && is a short-circuit meaning the second part should not run.

Reply:
You cloud just write a short function testing certain strings, or you just manually test it by hand within in your activated plugin.

Reply:
OK - so just check that it is all OK from the end user within the addon